modules/calculators/Pid.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file Pid.cpp
+ * @brief PIDを計算するクラス
+ * @author miyahara046
+ */
+
+#include "Pid.h"
+PidGain::PidGain(double _kp, double _ki, double _kd)
+  // pidゲインが負の値にならないようにする
+  : kp(_kp < 0 ? 0 : _kp), ki(_ki < 0 ? 0 : _ki), kd(_kd < 0 ? 0 : _kd)
+{
+}
+
+Pid::Pid(double _kp, double _ki, double _kd, double _targetValue, double _maxIntegral,
+         double _minIntegral)
+  : pidGain(_kp, _ki, _kd),
+    prevDeviation(0.0),
+    integral(0.0),
+    targetValue(_targetValue),
+    maxIntegral(_maxIntegral),
+    minIntegral(_minIntegral)
+{
+}
+
+void Pid::setPidGain(double _kp, double _ki, double _kd)
+{
+  // pidゲインが負の値にならないようにする
+  pidGain.kp = _kp < 0 ? 0 : _kp;
+  pidGain.ki = _ki < 0 ? 0 : _ki;
+  pidGain.kd = _kd < 0 ? 0 : _kd;
+}
+
+double Pid::calculatePid(double currentValue, double delta)
+{
+  // 0除算を避けるために0の場合はデフォルト周期0.01とする
```

Comment:
```suggestion
  // 0 除算を避けるために delta=0 の場合はデフォルト周期 0.01 とする
```
なにが 0 になるかを明確にした方が親切ではある